### PR TITLE
Perform a few tweaks to make the bench script work properly.

### PR DESCRIPTION
**Pull request by ppedrot**

### CODE REVIEW

**Comment by SkySkimmer** [comment]
Could you add the logs to the artifacts too?

**Comment by ppedrot** [comment]
@SkySkimmer please push a commit above this PR, I am not sure which logs you're referring to.

**Comment by ppedrot** [comment]
This should be ready then.

**Comment by Zimmi48** [comment]
The benchmarking infrastructure used to be documented in the wiki (https://github.com/coq/coq/wiki/Jenkins-(automated-benchmarking)). Can we get some documentation for the new infrastructure (can be a follow-up PR)? It could be in https://github.com/coq/coq/blob/master/dev/ci/README-developers.md. It should be linked from the contributing guide instead of the old wiki page (cf. the doc on the `needs: benchmarking` label at https://github.com/coq/coq/blob/master/CONTRIBUTING.md#understanding-reviewers-feedback). The old infrastructure had support for an overlay-like mechanism. What about the new infrastructure? Does it reuse the overlays from the CI system?

**Comment by ppedrot** [comment]
> The old infrastructure had support for an overlay-like mechanism

Did it? Apart from a way to use a different repo for the devs to be tested, I don't remember something of that kind...

**Comment by Zimmi48** [comment]
I'm talking about https://github.com/coq/coq/wiki/Jenkins-(automated-benchmarking)#benchmarking-with-overlays

**Comment by ppedrot** [comment]
Indeed, this is what I mentioned. Not really usable as an overlay system if you ask me, but the new bench infra does support the same variables, so you can do the same.

**Comment by ppedrot** [comment]
BTW I'd still recommend merging this quickly. It makes the bench usable, and both the docs and the nitpicks can be addressed later. (In the meantime I already have a few benches to run...)

File: dev/bench/gitlab.sh [hunk 1]
```diff
@@ -40,17 +40,18 @@ check_variable "coq_pr_number"
...
 check_variable "old_ocaml_switch"
 check_variable "old_coq_repository"
-old_coq_commit="609152467f4d717713b7ea700f5155fc9f341cd7"
 check_variable "old_coq_opam_archive_git_uri"
 check_variable "old_coq_opam_archive_git_branch"
 check_variable "num_of_iterations"
 check_variable "coq_opam_packages"
 
+new_coq_commit=$(git rev-parse HEAD^2)
+old_coq_commit=$(git merge-base HEAD^1 $new_coq_commit)
```
Comment (SkySkimmer) [inline dev/bench/gitlab.sh:53]: Why is this not 
```suggestion
old_coq_commit=$(git rev-parse HEAD^1)
```
?

Comment (ppedrot) [reply dev/bench/gitlab.sh:53]: Because otherwise you test with the diffs in master and you can get totally unrelated results.

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
@coqbot: merge now

**Pull request merged**